lib/tapioca/dsl/compilers/active_record_associations.rb

```diff
@@ -320,6 +321,36 @@ def declaration(reflection)
           end
         end
 
+        sig { params(reflection: ReflectionType).returns(T::Array[RBI::Comment]) }
+        def association_comments(reflection)
+          association = case reflection
+          when ActiveRecord::Reflection::HasOneReflection
+            "the-has-one-association"
+          when ActiveRecord::Reflection::HasManyReflection
+            "the-has-many-association"
+          when ActiveRecord::Reflection::HasAndBelongsToManyReflection
+            "the-has-and-belongs-to-many-association"
+          when ActiveRecord::Reflection::BelongsToReflection
+            "the-belongs-to-association"
+          when ActiveRecord::Reflection::ThroughReflection
+            delegate_reflection = reflection.send(:delegate_reflection)
+            declaration = declaration(delegate_reflection)
+            if declaration == "has_one"
+              "the-has-one-through-association"
+            else
+              "the-has-many-through-association"
+            end
+          end
+
+          if association
+            url = "https://guides.rubyonrails.org/association_basics.html##{association}"
+            association_name = association.sub(/^the-(.*)-association$/, '\1')
+            [RBI::Comment.new("🔗 [Rails guide for `#{association_name}` association](#{url})")]
```

Comment:
I really enjoyed @paracycle suggestion of comment before the link:

```
# This method is created at runtime by ActiveRecord on the `Post` class since `Post` has declared a `has_one` association to `User`.
```

I feel it's very explanatory of why the link is relevant and why I should go read this doc. Do you think it's doable?

Reply:
Ah I actually missed this part of the comment 🤦‍♂️ 
Yeah I think it's helpful. Will update 👍

Reply:
How about:

This method is created by ActiveRecord on the `Post` class because it declared `has_many :comments`.
🔗 [Rails guide for `has-many` association](https://guides.rubyonrails.org/association_basics.html#the-has-many-association)

This method is created by ActiveRecord on the `Post` class because it declared `has_many :commenters, through: :comments`.
🔗 [Rails guide for `has-many-through` association](https://guides.rubyonrails.org/association_basics.html#the-has-many-through-association)

```md
This method is created by ActiveRecord on the `Post` class because it declared `has_many :comments`.
🔗 [Rails guide for `has-many` association](https://guides.rubyonrails.org/association_basics.html#the-has-many-association)

This method is created by ActiveRecord on the `Post` class because it declared `has_many :commenters, through: :comments`.
🔗 [Rails guide for `has-many-through` association](https://guides.rubyonrails.org/association_basics.html#the-has-many-through-association)
```

Reply:
Yeah I like it 👍 

Nit: why is the association hyphenated in the link text: `has-many` instead of `has_many`?